node/tcp/src/helpers/config.rs

```diff
@@ -84,7 +84,7 @@ impl Default for Config {
             name: None,
             listener_ip: default_ip(),
             desired_listening_port: None,
-            allow_random_port: true,
+            allow_random_port: false,
```

Comment:
this belongs in a separate commit IMO

Reply:
Not sure what you mean? All the rest ARE the necessary test changes. I've split up this change from the test changes. Is that what you meant?

Reply:
sorry for not being clear; I meant the extension of the node creation methods, but I'm guessing those were necessary for the test adjustments

Reply:
Indeed. Still it makes sense to split of the actual fix from the test fixes.